client-js/evomaster-client-js/tests/unit/instrumentation/babel-code-test.ts

```diff
@@ -400,5 +400,163 @@ test("ternary throw", () => {
 });
 
 
+test("purity analysis 'and' and 'or' with literal/binary expression", () => {
```

Comment:
can you add tests to check out non-pure cases like:

assignment: eg, `if(x=42)`
increment/decrement: eg, `if(x++)`

Reply:
tests are added.